shared/js/content-scripts/element-hiding.js

```diff
@@ -7,124 +7,131 @@
+         * Iterate through collection of potentially blocked elements, comparing against list
+         * of blocked requests. In the main frame all we care about is iframes, but within iframes
+         * we also look for blocked scripts that may have loaded a nested iframe
+         */
+        locateBlockedFrames (requests) {
+            if (this.frameType === 'main') {
+                let i = this.foundFrames.length
+                while (i--) {
+                    let frame = this.foundFrames[i]
+                    if (requests.includes(frame.src)) {
+                        this.hideFrame(frame)
+                    }
+                }
+            } else if (this.frameType === 'topLevelFrame') {
+                let i = this.possibleTargets.length
+                while (i--) {
+                    let elem = this.possibleTargets[i]
+                    if (elem.src && requests.includes(elem.src)) {
+                        window.top.postMessage({frameId: this.frameId, type: 'hideFrame'}, this.mainFrameUrl)
+                        window.removeEventListener('message', this.frameListener)
+                        break
+                    }
+                }
             }
-        } else if (req.type === 'disable') {
-            window.removeEventListener('message', contentScript.frameListener)
         }
-    })
 
-    // Initialize event listener for interframe messaging
-    window.addEventListener('message', contentScript.frameListener)
+        /**
+         * Hide frames that were either themselves blocked, or that contain scripts
+         * or other frames that were blocked. Then traverse DOM upward, hiding
+         * parent selector if it only contains the blocked frame. Add class
+         * and remove event listeners so that other content scripts no longer
+         * interact with hidden frames
+         */
+        hideFrame (frame) {
+            frame.style.setProperty('display', 'none', 'important')
+            frame.hidden = true
+            frame.classList.add('ddg-hidden')
 
-    /**
-     * When document hits 'interactive' readystate, all DOM elements have
-     * been added to page and all requests for this document have been initiated.
-     * At this point, kick off process by requesting list of blocked requests from
-     * background page.
-     */
-    if (document.readyState === 'loading') {
-        document.addEventListener('readystatechange', () => {
-            if (document.readyState === 'interactive') {
-                chrome.runtime.sendMessage({hideElements: true, url: document.location.href, frame: contentScript.frameType})
+            if (frame.parentNode.childElementCount === 1) {
+                this.hideFrame(frame.parentNode)
             }
-        })
-    } else {
-        chrome.runtime.sendMessage({hideElements: true, url: document.location.href, frame: contentScript.frameType})
+        }
     }
+
+    // Instantiate content script
+    var contentScript = new ContentScript()
```

Comment:
eslint is throwing a no-unused-var error here - https://travis-ci.org/duckduckgo/duckduckgo-privacy-extension/builds/408297396.